src/cdk/overlay/position/connected-position-strategy.ts

```diff
@@ -428,14 +433,26 @@ export class ConnectedPositionStrategy implements PositionStrategy {
     this._onPositionChange.next(positionChange);
   }
 
-  /**
-   * Subtracts the amount that an element is overflowing on an axis from it's length.
-   */
+  /** Subtracts the amount that an element is overflowing on an axis from it's length. */
   private _subtractOverflows(length: number, ...overflows: number[]): number {
     return overflows.reduce((currentValue: number, currentOverflow: number) => {
       return currentValue - Math.max(currentOverflow, 0);
     }, length);
   }
+
+  /** Validates that the current position match the expected values. */
+  private _validatePositions(): void {
+    if (!this._preferredPositions.length) {
+      throw Error('ConnectedPositionStrategy: At least one position is required.');
+    }
+
+    this._preferredPositions.forEach(pair => {
```

Comment:
Add a TODO to remove this check once Angular's template type checker is good enough to not need it?